bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/BundleProcessor.java

```diff
@@ -214,6 +235,38 @@ private CSVMetadataHolder createMetadataHolder(AnnotationInstance csvMetadata, C
             maintainers = mh.maintainers;
         }
 
+        final var linksField = csvMetadata.value("links");
+        CSVMetadataHolder.Link[] links;
+        if (linksField != null) {
+            final var linkAnn = linksField.asNestedArray();
+            links = new CSVMetadataHolder.Link[linkAnn.length];
+            for (int i = 0; i < linkAnn.length; i++) {
+                links[i] = new CSVMetadataHolder.Link(
+                        ConfigurationUtils.annotationValueOrDefault(linkAnn[i], "name",
+                                AnnotationValue::asString, () -> null),
+                        ConfigurationUtils.annotationValueOrDefault(linkAnn[i], "url",
+                                AnnotationValue::asString, () -> null));
+            }
+        } else {
+            links = mh.links;
+        }
+
+        final var iconField = csvMetadata.value("icon");
+        CSVMetadataHolder.Icon[] icon;
+        if (iconField != null) {
+            final var iconAnn = iconField.asNestedArray();
+            icon = new CSVMetadataHolder.Icon[iconAnn.length];
+            for (int i = 0; i < iconAnn.length; i++) {
+                icon[i] = new CSVMetadataHolder.Icon(
+                        ConfigurationUtils.annotationValueOrDefault(iconAnn[i], "base64data",
```

Comment:
In particular, passing base64 data in an annotation is not the most user-friendly thing out there ;)